framework/Source/Kirakira.swift

```diff
@@ -10,9 +10,87 @@ import Foundation
 
 public class Kirakira: OperationGroup {
 
-    public enum ColorMode: Int {
+    public enum ColorMode: Int, Codable {
         case white = 0
         case random = 1
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            let rawValue = try container.decode(Int.self)
+            self = ColorMode(rawValue: rawValue) ?? .random
```

Comment:
I think we should throw an error instead of using the random type.

Reply:
+1 `ColorMode` can conform to Codable without `init(from decoder: Decoder)` and `func encode(to encoder: Encoder)` for free

If the caller expect to get `.random` on decoding failure, they can go with `(try? decode(ColorMode.self, ...)) ?? .random`